cmake/mode.common.cmake

```diff
@@ -24,9 +24,8 @@ add_compile_options(
   ${_supported_warnings})
 
 function(default_target_arch arch)
-  set(x86_instruction_sets i386 i686 x86_64)
-  if(CMAKE_SYSTEM_PROCESSOR IN_LIST x86_instruction_sets)
-    set(${arch} "westmere" PARENT_SCOPE)
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+    set(${arch} "broadwell" PARENT_SCOPE)
```

Comment:
Better to use x86-64-v3 which is a neutral term.

Reply:
Yes, but that means additional changes will be required to the scripts, as I'll need to add the -m pclmul somehow.

Reply:
I'm sure @tchaikov can find out how.

Reply:
```diff
diff --git a/cmake/mode.common.cmake b/cmake/mode.common.cmake
index 35e58415e9..a2b8a694ff 100644
--- a/cmake/mode.common.cmake
+++ b/cmake/mode.common.cmake
@@ -23,10 +23,11 @@ add_compile_options(
   "-Wimplicit-fallthrough"
   ${_supported_warnings})
 
-function(default_target_arch arch)
+function(default_target_arch arch target_features)
   set(x86_instruction_sets i386 i686 x86_64)
   if(CMAKE_SYSTEM_PROCESSOR IN_LIST x86_instruction_sets)
-    set(${arch} "westmere" PARENT_SCOPE)
+    set(${arch} "x86-64-v3" PARENT_SCOPE)
+    set(${target_features} "pclmul")
   elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
     # we always use intrinsics like vmull.p64 for speeding up crc32 calculations
     # on the aarch64 architectures, and they require the crypto extension, so
@@ -85,10 +86,23 @@ endfunction()
 
 add_compile_options("-ffile-prefix-map=${CMAKE_BINARY_DIR}=.")
 
-default_target_arch(target_arch)
+default_target_arch(target_arch target_features)
 if(target_arch)
-  add_compile_options("-march=${target_arch}")
+  add_compile_options(
+    "-march=${target_arch}"
+    "-target-features=pclmul")
 endif()
+foreach(target_features ${target_features})
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # see https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
+    add_compile_options(
+      "-m${target_feature}")
+  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    add_compile_options(
+      "-Xclang" "-target-feature"
+      "-Xclang" "+${feature}")
+  endif()
+endforeach()
 
 add_compile_options("SHELL:-Xclang -fexperimental-assignment-tracking=disabled")
 
```

please let me know if anything else.

Reply:
Wow. Not sure if it's better than just use 'broadwell'.

Reply:
Does clang not support -mpclmul? Usually it follows gcc.

Reply:
I'm sure it does. Now I cannot reproduce the perf simple query improved results though...

BTW, I was going to submit a similar PR for aarch64 to move to 8.2, but I can't easily test that.

Reply:
> Does clang not support -mpclmul? Usually it follows gcc.

indeed. so no need to check for the compiler identifier. and we can always pass `-m${target_feature}"` to compiler.